disruptive/resources/claim.py

```diff
@@ -0,0 +1,197 @@
+        dry_run : bool, optional
+            Test your claim request during development.
+            No kits or devices will be claimed.
+        **kwargs
+            Arbitrary keyword arguments.
+            See the :ref:`Configuration <configuration>` page.
+
+        Returns
+        -------
+        devices : list[Claim.Device]
+            List of successfully claimed devices.
+        errors : list[ClaimError]
+            List of errors that occured during claiming.
+            If list is empty, all devices were claimed successfully.
+
+        """
+
+        url = f'/projects/{project_id}/devices:claim'
+        url += f'?dryRun={str(dry_run).lower()}'
+
+        body = {}
+        if kit_ids is not None:
+            body['kitIds'] = kit_ids
+        if device_ids is not None:
+            body['deviceIds'] = device_ids
+
+        res = dtrequests.DTRequest.post(url, body=body, **kwargs)
+
+        return (
+            [Claim.Device(d) for d in res['claimedDevices']],
+            Claim._parse_claim_errors(res['claimErrors']),
+        )
+
+    @staticmethod
+    def _parse_claim_errors(res_errors: dict) -> List[Exception]:
+        errors: List[Exception] = []
+        for error in res_errors['devices'] + res_errors['kits']:
+            if error['code'] == 'ALREADY_CLAIMED':
+                errors.append(dterrors.ClaimErrorDeviceAlreadyClaimed(error))
+            elif error['code'] == 'NOT_FOUND' and 'deviceId' in error:
+                errors.append(dterrors.ClaimErrorDeviceNotFound(error))
+            elif error['code'] == 'NOT_FOUND' and 'kitId' in error:
+                errors.append(dterrors.ClaimErrorKitNotFound(error))
+            else:
+                errors.append(dterrors.ClaimError(error))
+        return errors
+
+    def _resolve_type(self, claim: dict) -> Claim.Kit | Claim.Device:
+        """
+        Return either Kit or Device depending on type.
+
+        Parameters
+        ----------
+        claim : dict
+            Unmodified claim response dictionary.
+
+        Returns
+        -------
+        claim_item : Claim.Kit | Claim.Device
+            Either a Kit or Device object depending on type.
+
+        """
+
+        if claim['type'] == Claim.KIT:
+            return Claim.Kit(claim['kit'])
+        elif claim['type'] == Claim.DEVICE:
+            return Claim.Device(claim['device'])
+        else:
+            raise KeyError(f'unknown claim type {claim["type"]}')
+
+    class Device(dtoutputs.OutputBase):
+        """
+        Namespacing type for a claimed device.
+
+        """
+
+        def __init__(self, device: dict) -> None:
+            """
+            Constructs the claimed Device object from raw response.
+
+            """
+
+            # Inherit from OutputBase parent.
+            dtoutputs.OutputBase.__init__(self, device)
+
+            # Unpack attributes from raw response dictionary.
+            self.device_id: str = device['deviceId']
```

Comment:
If possible, I think these fields should also be documented?

Reply:
Correct, I've forgotten the `Attributes` section in the `Device` docstring and will add it.